quantus/functions/similarity_func.py

```diff
@@ -66,14 +85,20 @@ def correlation_kendall_tau(a: np.array, b: np.array, **kwargs) -> float:
          The first array to use for similarity scoring.
     b: np.ndarray
          The second array to use for similarity scoring.
+    batched: bool
+         True if arrays are batched. Arrays are expected to be 2D (B x F), where B is batch size and F is the number of features
     kwargs: optional
         Keyword arguments.
 
     Returns
     -------
-    float
-        The similarity score.
+    Union[float, np.array]
+        The similarity score or a batch of similarity scores.
     """
+    if batched:
+        assert len(a.shape) == 2 and len(b.shape) == 2, "Batched arrays must be 2D"
+        # No support for axis currently, so just iterating over the batch
+        return np.array([scipy.stats.kendalltau(a_i, b_i)[0] for a_i, b_i in zip(a, b)])
```

Comment:
Mb we could use `np.vectorize` (https://numpy.org/doc/stable/reference/generated/numpy.vectorize.html) for this one?

Reply:
could be used but I also like the simplicity of @davor10105's suggestion!